src/javascripts/components/cookies-page.js

```diff
@@ -0,0 +1,167 @@
+import { getConsentCookie, setConsentCookie } from '../cookie-functions.js'
+import common from 'govuk-frontend/govuk/common'
+
+const nodeListForEach = common.nodeListForEach
+
+const cookieFormGroupSelector = '.js-cookies-page-form-group'
+const errorSummarySelector = '.js-cookies-page-error-summary'
+const saveButtonSelector = '.js-cookies-page-save'
+const successNotificationSelector = '.js-cookies-page-success'
+
+function CookiesPage ($module) {
+  this.$module = $module
+}
+
+CookiesPage.prototype.init = function () {
+  this.$cookiePage = this.$module
+
+  if (!this.$cookiePage) {
+    return
+  }
+
+  this.$cookieFormGroups = this.$cookiePage.querySelectorAll(cookieFormGroupSelector)
+  this.$errorSummary = this.$cookiePage.querySelector(errorSummarySelector)
+  this.$saveButton = this.$cookiePage.querySelector(saveButtonSelector)
+  this.$successNotification = this.$cookiePage.querySelector(successNotificationSelector)
+
+  this.userCookiePreferences = getConsentCookie()
+
+  nodeListForEach(this.$cookieFormGroups, function ($cookieFormGroup) {
+    this.showUserPreference($cookieFormGroup, this.userCookiePreferences)
+    $cookieFormGroup.removeAttribute('hidden')
+  }.bind(this))
+
+  this.$saveButton.addEventListener('click', this.savePreferences.bind(this))
+  this.$saveButton.removeAttribute('hidden')
+}
+
+CookiesPage.prototype.savePreferences = function () {
+  var preferences = {}
+
+  nodeListForEach(this.$cookieFormGroups, function ($cookieFormGroup) {
+    var cookieType = this.getCookieType($cookieFormGroup)
+    var selected = $cookieFormGroup.querySelector(':checked')
+
+    if (selected === null) {
+      this.showErrorMessage($cookieFormGroup)
+      return
+    } else {
+      this.hideErrorMessage($cookieFormGroup)
+    }
+
+    preferences[cookieType] = selected.value === 'yes'
```

Comment:
Feels slightly odd to combine a `return` and an `else` like this when there's further code after the if/else block as well – effectively `this.hideErrorMessage($cookieFormGroup)` and `preferences[cookieType] = selected.value === 'yes'` are part of the same 'flow' but are split up.

It might be clearer to lose the `else`?

```suggestion
    if (selected === null) {
      this.showErrorMessage($cookieFormGroup)
      return
    }

    this.hideErrorMessage($cookieFormGroup)
    preferences[cookieType] = selected.value === 'yes'
```